src/textual/widgets/_selection_list.py

```diff
@@ -0,0 +1,655 @@
+            disabled: The initial enabled/disabled state. Enabled by default.
+        """
+        if isinstance(prompt, str):
+            prompt = Text.from_markup(prompt)
+        super().__init__(prompt.split()[0], id, disabled)
+        self._value: SelectionType = value
+        """The value associated with the selection."""
+        self._initial_state: bool = initial_state
+        """The initial selected state for the selection."""
+
+    @property
+    def value(self) -> SelectionType:
+        """The value for this selection."""
+        return self._value
+
+    @property
+    def initial_state(self) -> bool:
+        """The initial selected state for the selection."""
+        return self._initial_state
+
+
+class SelectionList(Generic[SelectionType], OptionList):
+    """A vertical selection list that allows making multiple selections."""
+
+    BINDINGS = [Binding("space", "select")]
+    """
+    | Key(s) | Description |
+    | :- | :- |
+    | space | Toggle the state of the highlighted selection. |
+    """
+
+    COMPONENT_CLASSES: ClassVar[set[str]] = {
+        "selection-list--button",
+        "selection-list--button-selected",
+        "selection-list--button-highlighted",
+        "selection-list--button-selected-highlighted",
+    }
+    """
+    | Class | Description |
+    | :- | :- |
+    | `selection-list--button` | Target the default button style. |
+    | `selection-list--button-selected` | Target a selected button style. |
+    | `selection-list--button-highlighted` | Target a highlighted button style. |
+    | `selection-list--button-selected-highlighted` | Target a highlighted selected button style. |
+    """
+
+    DEFAULT_CSS = """
+    SelectionList > .selection-list--button {
+        text-style: bold;
+        background: $foreground 15%;
+    }
+
+    SelectionList:focus > .selection-list--button {
+        text-style: bold;
+        background: $foreground 25%;
+    }
+
+    SelectionList > .selection-list--button-highlighted {
+        text-style: bold;
+        background: $foreground 15%;
+    }
+
+    SelectionList:focus > .selection-list--button-highlighted {
+        text-style: bold;
+        background: $foreground 25%;
+    }
+
+    SelectionList > .selection-list--button-selected {
+        text-style: bold;
+        background: $foreground 15%;
+    }
+
+    SelectionList:focus > .selection-list--button-selected {
+        text-style: bold;
+        color: $success;
+        background: $foreground 25%;
+    }
+
+    SelectionList > .selection-list--button-selected-highlighted {
+        text-style: bold;
+        color: $success;
+        background: $foreground 15%;
+    }
+
+    SelectionList:focus > .selection-list--button-selected-highlighted {
+        text-style: bold;
+        color: $success;
+        background: $foreground 25%;
+    }
+    """
+
+    class SelectionMessage(Generic[MessageSelectionType], Message):
+        """Base class for all selection messages."""
```

Comment:
I find this misleading because there is a "selection message" below that doesn't inherit from this.

Reply:
The one below doesn't deal with a selection.

Reply:
gotcha